### ISSUE TITLE
PROCESS CHANGE: add new ready-to-work and not-ready-to-work labels

### ISSUE THREAD

**Issue by Cali0707** [knative/community#1403]
## Expected Benefits
By adding ready-to-work and not-ready-to-work labels to issues, contributors will have a much easier time finding issues to work on. We currently are marking issues "ready to work" in github projects, but not every issue is a part of a project and not every contributor will look at the projects to see if it is ready to work.

## Expected Costs
This will incur a small cost on the maintainers of repos as it will be another label to add to issues during the triage process. There is also a chance that the labels could get out of sync with the github projects (unless we found/made a prow plugin to automatically keep those in sync).

## Timeframe
Adding the labels can be done very quickly, and then the roll out of actually using the labels in individual repos will be up to the maintainers of each repo. Finding/creating a prow plugin to keep labels in sync with the github projects would take longer, but even if we want to pursue that path I don't think it would be a blocker to rolling out the new labels.

I am willing to add any necessary labels, and pursue the prow plugin path if needed.

**Comment by Cali0707** [comment]
@aliok @pierDipi @nainaz

**Comment by dprotaso** [comment]
> but not every issue is a part of a project

This seems to be the crux of the problem no?

>  not every contributor will look at the projects to see if it is ready to work.

I'm curious how do you envision people discovering the `ready-to-work` label?

**Comment by Cali0707** [comment]
> > but not every issue is a part of a project
> 
> This seems to be the crux of the problem no?

Yeah, I think so. However, I don't think that making every issue a part of a project would be a good fix to the problem. There are cases where there is a one off issue, or there are only one or two issues for a feature. In these cases I think making a project would likely be overkill.
> 
> > not every contributor will look at the projects to see if it is ready to work.
> 
> I'm curious how do you envision people discovering the `ready-to-work` label?

I was thinking linking to a github issue query similar to https://github.com/issues?q=is%3Aissue+archived%3Afalse+user%3Aknative+is%3Aopen+label%3A%22good+first+issue%22 but with the newly created label.

WDYT @dprotaso ?

**Comment by dprotaso** [comment]
So I think we have what you want already in place - the `help wanted` label is used across repositories

And this feeds into a CNCF website contributors can go to https://clotributor.dev/search?project=knative&ts_query_web=knative 

What's cool about the UI it lets you filter good first issues as well

Though it looks like the `Area` tab only lets you filter for the docs repo - cc @aliok how do you add more 'Area's eg. Serving


cc @knative/technical-oversight-committee for input

**Comment by dprotaso** [comment]
Given the label I think the next step could be updating repo readmes to link to the CLOtributor page

**Comment by Cali0707** [comment]
> Given the label I think the next step could be updating repo readmes to link to the CLOtributor page. And we should definitely call it out on the website

I can work on this next week!
/assign

**Comment by aliok** [comment]
I personally like how this discussion evolved :) Looks like we just need to make use of `help-wanted` and `good-first-issue` labels more and also raise awareness to provide instructions in these issues for any newcomers.

To answer this question:

> Though it looks like the Area tab only lets you filter for the docs repo - cc @aliok how do you add more 'Area's eg. Serving

https://github.com/cncf/clotributor#labels-with-special-meaning

There's no other area right now supported by CLOtributor.

> Area
> - docs: the issue has a label that contains the string docs or documentation.

**Comment by dprotaso** [comment]
@Cali0707 is there anything else remaining for this issue?

**Comment by Cali0707** [comment]
Ah sorry @dprotaso this got lost in my todo list, there are still some repos that need the CLOtributor link added. I'll try and wrap this up tomorrow.
